syntax = "proto3";

package user;

import "google/protobuf/wrappers.proto";
import "google/protobuf/empty.proto";

message User {
    int64 id = 1;
    string name = 2;
    string created_at = 3;
    string updated_at = 4;
  }

service UserService {
    rpc CreateUser(User) returns (User);
    rpc ReadUser(google.protobuf.Int64Value) returns (User);
    rpc UpdateUser(User) returns (User);
    rpc DeleteUser(google.protobuf.Int64Value) returns (User);
    rpc ListUsers(google.protobuf.Empty) returns (stream User);
}